File: Programming Basics/JavaScript/Nested loops/Lab/travelling.js
function solve(input) {
    let index = 0;
    let destination = input[index];
    while (destination != "End") {
        const minBudget = Number(input[++index]);

        let savedMoney = 0;
        while (savedMoney < minBudget) savedMoney += Number(input[++index]);

        console.log(`Going to ${destination}!`);
        destination = input[++index];
    }
}

File: Programming Basics/C++/Nested loops/Labs/Travelling/Travelling.cpp
#include <iostream>
using namespace std;

int main()
{
    string destination;
    while (cin >> destination && destination != "End")
    {
        double minBudget;
        cin >> minBudget;

        double savedMoney = 0;
        while (savedMoney < minBudget)
        {
            double currentAmount;
            cin >> currentAmount;

            savedMoney += currentAmount;
        }

        cout << "Going to " << destination << "!" << endl;
    }
}
